app/frontend/actions.ts

```diff
@@ -186,26 +193,50 @@ export default (store: Store) => {
       stopLoadingAction(state)
     }
 
-    // TODO: implement tx differenciation here and drop the txConfirmType
+    /*
+    REFACTOR:
+    Drop: `keepConfirmationDialogOpen, isCrossAccount` as those values can
+    be inferred
+    */
+
+    const isTxBetweenAccounts =
+      state.activeMainTab === MainTabs.ACCOUNT && txConfirmType === TxType.SEND_ADA
 
-    const isTxBetweenAccounts = state.activeMainTab === MainTabs.ACCOUNT && txConfirmType === 'send'
     // TODO: refactor
     const keepConfirmationDialogOpen =
       isTxBetweenAccounts ||
-      txConfirmType === 'convert' ||
-      txConfirmType === 'withdraw' ||
-      txConfirmType === 'deregisterStakeKey'
+      txConfirmType === TxType.CONVERT_LEGACY ||
+      txConfirmType === TxType.WITHDRAW ||
+      txConfirmType === TxType.DEREGISTER_STAKE_KEY
 
     setState({
       shouldShowConfirmTransactionDialog: true,
-      txConfirmType: isTxBetweenAccounts ? 'crossAccount' : txConfirmType,
```

Comment:
I believe that "crossAccount" should not be `txConfirmType` so I moved it to a separate property that I would also ideally remove later on, and only make it computed